### PR TITLE
feat: make refLabel editable, otherwise refLabelDefault is used

### PR DESCRIPTION
**Pull request by mswertz**
motivation: change how a ref is shown using a template.

- [x] docs
- [x] schema editor enables editing of refLabel
- [x] csv import/export supports changing the refLabel
- [x] in table explorer refLabel is used, otherwise refLabelDefault is used
- [x] in ref input refLabel is used, otherwise refLabelDefault is used
- [x] fix tests

### CODE REVIEW

File: apps/molgenis-components/src/client/client.ts [hunk 1]
```diff
@@ -82,7 +82,8 @@ export default {
           properties,
           schemaMetaData,
           myAxios,
-          schemaNameCache
+          schemaNameCache,
+          1
```
Comment (connoratrug) [inline apps/molgenis-components/src/client/client.ts:86]: que ? , '1'

File: apps/molgenis-components/src/client/client.ts [hunk 2]
```diff
@@ -116,7 +117,8 @@ export default {
             },
             schemaMetaData,
             myAxios,
-            schemaNameCache
+            schemaNameCache,
+            1
```
Comment (connoratrug) [inline apps/molgenis-components/src/client/client.ts:121]: same

File: apps/molgenis-components/src/client/client.ts (hunk 1 above)
Comment (mswertz) [reply apps/molgenis-components/src/client/client.ts:86]: you could have checked the variable name function that is being called. 1 means '1 level deep' which is what the current default is; but if you increase this you will be able to expand the graphql to expand refs with N levels.

Comment (connoratrug) [reply apps/molgenis-components/src/client/client.ts:86]: suggest to use a named constant

File: apps/molgenis-components/src/client/client.ts [hunk 3]
```diff
@@ -375,61 +379,48 @@ const request = async (url: string, graphql: string, variables?: any) => {
 };
 
 /**
- *
+ * @pram {String} schemaName - schema where initial table is in
```
Comment (chinook25) [inline apps/molgenis-components/src/client/client.ts:382]: pram :p

File: apps/molgenis-components/src/components/tables/RenderValue.vue [hunk 4]
```diff
@@ -7,7 +7,7 @@
       />
       <EmailDisplay v-else-if="col.columnType === 'EMAIL_ARRAY'" :data="val" />
       <span v-else>
-        {{ col.refLabel ? applyJsTemplate(col.refLabel, val) : val }}
+        {{ getLabel(col) ? applyJsTemplate(getLabel(col), val) : val }}
```
Comment (chinook25) [inline apps/molgenis-components/src/components/tables/RenderValue.vue:10]: I think this ternary can also go into a function, in order to have less logic in the html

File: apps/schema/src/components/Schema.vue [hunk 5]
```diff
@@ -200,7 +200,7 @@ export default {
       this.loading = true;
       request(
         "graphql",
-        "{_session{schemas,roles}_schema{name,tables{name,tableType,inherit,externalSchema,labels{locale,value},descriptions{locale,value},semantics,columns{name,labels{locale,value},table,position,columnType,inherited,key,refSchema,refTable,refLink,refBack,required,readonly,descriptions{locale,value},semantics,validation,visible} } } }"
+        "{_session{schemas,roles}_schema{name,tables{name,tableType,inherit,externalSchema,labels{locale,value},descriptions{locale,value},semantics,columns{name,labels{locale,value},table,position,columnType,inherited,key,refSchema,refTable,refLink,refBack,refLabel,refLabelDefault,required,readonly,descriptions{locale,value},semantics,validation,visible} } } }"
```
Comment (chinook25) [inline apps/schema/src/components/Schema.vue:203]: Using gql to have this layouted would be nice

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/Column.java [hunk 6]
```diff
@@ -510,23 +510,20 @@ public ColumnType getPrimitiveColumnType() {
     } else return getColumnType().getBaseType();
   }
 
-  public String getRefLabelIfSet() {
+  public String getRefLabel() {
     return this.refLabel;
   }
 
-  public String getRefLabel() {
+  public String getRefLabelDefault() {
```
Comment (chinook25) [inline backend/molgenis-emx2/src/main/java/org/molgenis/emx2/Column.java:517]: Way better names, nice :D

**Review by chinook25 — APPROVED** [review]
Looks good, and seems to work. 
I wonder, would it be nice to show both the id and the label in the table view. Currently it show the column id there, while it shows the label when adding/editing a row. I would imagine it would be nice to have the column header show something like 'id (label)'  (or the other way around).

**Pull request merged**